src/apps/packet_filter/packet_filter.lua

```diff
@@ -354,17 +354,18 @@ local function generateConformFunctionString(rules)
    T"return function(buffer, size)"
    T:indent()
 
-   if rules.state_track then
-      conntrack.define(rules.state_track)
+   if options.state_track then
+      conntrack.define(options.state_track)
       T"if (function(buffer, size)"
       T:indent()
    end
-   
-   if rules.state_check then
-      conntrack.define (rules.state_check)
-      T('if state_pass("', rules.state_check, '", buffer) then return true end')
+
+   if options.state_check then
+      conntrack.define (options.state_check)
+      T('if state_pass("', options.state_check, '", buffer) then return true end')
    end
 
+   local rules = options.rules or {}
```

Comment:
@lukego I could make this more strict, but since we can not reliably check for the correct type of the rules table anyways... We say it must be a *list*, which we might want to call "array" because that's the closest thing PIL describes. We can't AFAIK easily check that rules is indeed an array though.

I think the current code is OK: The manual says do it that way and the code handles user errors on a range between forgiving (default to empty rule array) to undefined (I don't know what happens when the table has e.g. string keys, or non-table values, it might ignore those keys or throw up. I don't think that in this specific case there is imminent danger for an API user to rely on undefined behavior).